### ISSUE TITLE
ListItemCollection is 0 for SharePoint list

### ISSUE THREAD

**Issue by Xeyow** [vgrem/Office365-REST-Python-Client#779]
Hello.

It looks like Microsoft no longer allows logging into SharePoint using my username and password, so I had to create an Azure App with the following permissions: **Sites.ReadWrite.All** and **Sites.Selected**. Those two permissions should allow the app to read/write items from/to a SharePoint list. I've also created a client_secret (that will be used in Python afterwards)

After creating the app, I went to: _http://company.sharepoint/site/nameofsite/_layouts/15/appinv.aspx_ to invite that app for which I gave the following permissions:

```
<AppPermissionRequests AllowAppOnlyPolicy="true">
    <AppPermissionRequest Scope="https://company.sharepoint.com/sites/nameofsite" Right="FullControl" />
    <AppPermissionRequest Scope="https://company.sharepoint.com/sites/nameofsite/lists/*" Right="Read,Write" />
</AppPermissionRequests>

```

In Python I have the following code (app_settings is correctly defined)

```
ctx_auth = AuthenticationContext(app_settings['url'])
if ctx_auth.acquire_token_for_app(client_id=app_settings['client_id'], client_secret=app_settings['client_secret']):
    ctx = ClientContext(app_settings['url'], ctx_auth)
    web = ctx.web
    lists = web.lists.get().execute_query()
    target_list = ctx.web.lists.get_by_title("Licenses").get().execute_query()
    print("List title: {0}".format(target_list.title))
    all_items_of_list = target_list.items.top(100)
    ctx.load(all_items_of_list).execute_query()
    # Access and print item properties
    for item in all_items_of_list:
        print("Item Title: {0}".format(item.properties["Title"]))
```

The first print (the one that prints the **targe_list.title**) works just fine as I can see in debug mode the variable **target_list.title** having the correct value. Now, the following part should grab the first 100 items of that list, and print their title... This doesn't work, as the **items = {ListItemCollection: 0}[]**, although the **item_count** of that list is equal to 1910 (again, I can see the value of the variable in debug mode).

In conclusion: although I can log in into my site, "open" the SharePoint List and grab its name (also, the number of items is correct), I can not read any item. What am I missing?

Thanks!

**Comment by vgrem** [comment]
Greetings,


```
AppPermissionRequest Scope="https://company.sharepoint.com/sites/nameofsite" Right="FullControl" />
                                                                  ^^^^^^^^^^
                                                                  custom scope name, namely nameofsite is not supported
```

```
<AppPermissionRequest Scope="https://company.sharepoint.com/sites/nameofsite/lists/*" Right="Read,Write" />
                                                                  ^^^^^^^^^^^^^
                                                                  
```


per [documentation](https://learn.microsoft.com/en-us/sharepoint/dev/sp-add-ins/add-in-permissions-in-sharepoint) `scope` is expected to be one of the folowing values:

- Tenancy	`http://sharepoint/content/tenant `
- Site Collection	`http://sharepoint/content/sitecollection`
- Website	`http://sharepoint/content/sitecollection/web`	
- List	`http://sharepoint/content/sitecollection/web/list`
